### DISCUSSION TITLE
Cannot read properties of undefined (reading 'defaultPrevented').

### DISCUSSION THREAD

**Discussion by meshv-p** [tailwindlabs/headlessui#1945] [Help]
Hey all, I am trying to fetch item when user type the item name and show to combobox. But..
What I want is after showing list of item, if user press enter on item then I will redirect to item  page. However, I can't access that item in lieu of getting error like :-
### Cannot read properties of undefined (reading 'defaultPrevented')

![image](https://user-images.githubusercontent.com/77038149/197199504-3b7e1d75-6db4-4d85-b969-53d1404b6b1e.png)

Here is my code

![image](https://user-images.githubusercontent.com/77038149/197200071-cdc12925-d5b0-4a70-9fdc-9779517c33d5.png)

**Comment by auchynnikau** [comment]
One of the solutions I see there is that you can wrap your code into condition where you'll check if `item` exist or it's not empty. As the result you component will renders only when `item` exists and you can avoid this error.

**Reply by meshv-p** [reply]
But, item is there. as I am applying condition of item.length != 0... Still getting this error.

**Reply by auchynnikau** [reply]
Can you provide more screenshots of your code, to understand the whole picture. I'd like to understand where `item` is declared.

**Reply by meshv-p** [reply]
## Here I am setting items 
![image](https://user-images.githubusercontent.com/77038149/199238347-445dd805-8606-4079-aead-886274d4feae.png)


## Here I am populating items
![image](https://user-images.githubusercontent.com/77038149/199238493-bc94046b-23f9-4983-922d-7dc9d7952f55.png)

**Comment by auchynnikau** [comment]
Alright, and what is the console output of first argument in `onChange` method?
